enhancements/cloud-integration/infrastructure-external-platform-type.md

```diff
@@ -0,0 +1,432 @@
+  however users are experiencing a lot of difficulties on this path due to necessity of, literally, reverse engineering,
+  manual management of generic MAPI controllers and so on.
+
+- As a cloud provider whose platform is not integrated into OpenShift,
+  I want to have the Cloud Controller Manager for my infrastructure running in OpenShift from the initial install.
+  Having a platform type that allows for the addition of operators or components which perform platform-specific functionality would help
+  me to create better integrations between OpenShift and my infrastructure platform.
+
+### Goals
+
+- Reduce the necessity to make changes in "openshift/api", "openshift/library-go" and dependant infrastructure related components during basic cloud-provider integration with OpenShift
+- Make cloud-provider integration process more accessible and simple to external contributors as well as for RH engineers
+- Provide a top level description of the operation mode for affected components at the first steps
+- Introduce a somewhat "neutral" platform type, which would serve as a signal about underlying cloud-provider presence
+
+### Non-Goals
+
+- describe concrete delivery mechanisms for cloud-provider specific components
+- cover new infrastructure provider enablement from RHCOS side
+- describe specific changes for each affected component, aside no-op reaction to thew new "External" platform type
+
+## Proposal
+
+Extend the [PlatformType](https://github.com/openshift/api/blob/51f399230d604fa013c2bb341040c4ad126e7309/config/v1/types_infrastructure.go#L128) "enum"
+with a special value "External". Which would be similar to the "None" value during initial phase, but alongside adding support of this across
+infrastructure related components it would change its semantic to signal when OpenShift is running on an infrastructure platform
+that will receive supplemental components from the infrastructure provider.
+
+### Implementation phases
+
+For make such change implementation process more smooth, it would be natural to split this activity to phases.
+The proposed phases breakdown can be found in [OCPPLAN-9429](https://issues.redhat.com/browse/OCPPLAN-9429).
+
+This document intended to describe initial phases of this project, proposed initial course of action:
+
+1. Update "openshift/api" with adding "External" PlatformType
+2. Ensure that all Red Hat operator tolerate the "External" platform and treat the same as "None" platform
+
+Next phase out of the scope for this EP:
+
+3. Modify operators for doing specific things when seeing "External" platform. Briefly it would be described down below, where possible for the moment.
+
+### Future work
+
+There are a number of topics in this area which would be wise to defer for upcoming phases, namely:
+
+1. Define missing capabilities and its concrete behaviour, for example add a "capability" for machine-api
+2. Precisely define reaction of the operators listed below for the "External" platform type 
```

Comment:
How do we know the current proposal will solve the issue if we don't explore this?

Reply:
We did explore this. Here is just a notion, that we do not want to describe components behaviour change in details within this EP.

Reply:
As someone reading this without the same context as you/the others involved in the project, I don't know that from reading the enhancement. You should make sure that's obvious and perhaps link out to anywhere that folks can read more about this. 

IMO I would expect the enhancement to paint the whole picture rather than just the first phase but maybe that's asking too much 🤔